### ISSUE TITLE
Your Android Demo project cann't compile at all, guys.

### ISSUE THREAD

**Issue by chriszeng87** [artoolkit/artoolkit5#383]
Error:(189, 36) error: cannot find symbol variable frameWidth
Error:(189, 49) error: cannot find symbol variable frameHeight
Error:(351, 15) error: cannot find symbol method isNativeInited()
Error:(345, 36) error: method startWithPushedVideo in class ARToolKit cannot be applied to given types;
required: int,int,String,String,int,boolean
found: int,int,<null>,int,boolean
reason: actual and formal argument lists differ in length

**Comment by chriszeng87** [comment]
Thank you. After switching to your recommended repo:
1. When I execute artoolkit5/android/build.sh, everything is OK.
2. When I execute artoolkit5/android/build_native_examples.sh, I got this error:
jni/../../lib/SRC/Eden/EdenSurfaces.c:395: error: undefined reference to 'glStateCacheEnableClientStateNormalArray'
jni/../../lib/SRC/Eden/EdenSurfaces.c:396: error: undefined reference to 'glStateCacheClientActiveTexture'
jni/../../lib/SRC/Eden/EdenSurfaces.c:398: error: undefined reference to 'glStateCacheEnableClientStateTexCoordArray'
jni/../../lib/SRC/Eden/EdenSurfaces.c:400: error: undefined reference to 'glStateCacheBindTexture2D'
jni/../../lib/SRC/Eden/EdenSurfaces.c:401: error: undefined reference to 'glStateCacheTexEnvMode'
jni/../../lib/SRC/Eden/EdenSurfaces.c:402: error: undefined reference to 'glStateCacheEnableTex2D'
jni/../../lib/SRC/Eden/EdenUtil.c:102: error: undefined reference to 'tcgetattr'
jni/../../lib/SRC/Eden/EdenUtil.c:107: error: undefined reference to 'tcsetattr'
jni/../../lib/SRC/Eden/EdenUtil.c:143: error: undefined reference to 'tcsetattr'
clang++: error: linker command failed with exit code 1 (use -v to see invocation)
make: *** [obj/local/armeabi-v7a/libARWrapperNativeCarsExample.so] Error 1
from /Users/qyvideo/work/deep_learning/artoolkit5: going to AndroidStudioProjects/ARNativeES1Proj/aRNativeES1
Android NDK: WARNING: APP_PLATFORM android-15 is higher than android:minSdkVersion 1 in ./AndroidManifest.xml. NDK binaries will *not* be comptible with devices older than android-15. See https://android.googlesource.com/platform/ndk/+/master/docs/user/common_problems.md for more information.    
Android NDK: ERROR:jni/Android.mk:util: LOCAL_SRC_FILES points to a missing file    
Android NDK: Check that jni/../../../../../../android/obj/local/armeabi-v7a/libutil.a exists  or that its path is correct   
/Users/qyvideo/work/android-ndk-r15c/build/core/prebuilt-library.mk:45: *** Android NDK: Aborting    .  Stop.
from /Users/qyvideo/work/deep_learning/artoolkit5: going to AndroidStudioProjects/ARNativeProj/aRNative
Android NDK: WARNING: APP_PLATFORM android-15 is higher than android:minSdkVersion 1 in ./AndroidManifest.xml. NDK binaries will *not* be comptible with devices older than android-15. See https://android.googlesource.com/platform/ndk/+/master/docs/user/common_problems.md for more information.    
Android NDK: ERROR:jni/Android.mk:util: LOCAL_SRC_FILES points to a missing file    
Android NDK: Check that jni/../../../../../../android/obj/local/armeabi-v7a/libutil.a exists  or that its path is correct   
/Users/qyvideo/work/android-ndk-r15c/build/core/prebuilt-library.mk:45: *** Android NDK: Aborting    .  Stop.
from /Users/qyvideo/work/deep_learning/artoolkit5: going to AndroidStudioProjects/ARNativeOSGProj/aRNativeOSG
Android NDK: WARNING: APP_PLATFORM android-15 is higher than android:minSdkVersion 1 in ./AndroidManifest.xml. NDK binaries will *not* be comptible with devices older than android-15. See https://android.googlesource.com/platform/ndk/+/master/docs/user/common_problems.md for more information.    
 Any suggestions?

**Comment by ThorstenBux** [comment]
Yes, unfortunately, the native build is broken. https://github.com/artoolkitx/artoolkit5/issues/120
You need to use the none native part for now.

**Comment by chriszeng87** [comment]
Could you fix it? Thanks!

**Comment by RobMoloney** [comment]
I'm getting the same issue.
I noticed that glStateCache.c and glStateCache2.c are not being compiled.
They are not added to the Make file in artoolkit/lib/SRC/Gl/Makefile.in  
https://github.com/artoolkitx/artoolkit5/issues/144

I tried adding them and not I get this error, looks like I'm missing something in the config.
gcc -O3 -fPIC -march=core2 -DHAVE_NFT=1 -pthread -I/usr/include/gstreamer-0.10 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/libxml2  -I../../../include/linux-x86_64 -I../../../include    -c -o glStateCache.o glStateCache.c
In file included from glStateCache.c:38:0:
../../../include/glStateCache.h:51:31: fatal error: OpenGLES/ES1/gl.h: No such file or directory

**Comment by willchamberlain** [comment]
Hello;  could anyone tell me the last commit which _was_ compiling? I'll have a look at the native build later this week, but for today I'd like to just revert to a version and compile, to show someone.

**Comment by ThorstenBux** [comment]
You can easily get started with artoolkitX. Everything is working there and you can use the provided examples: https://github.com/artoolkitx/artoolkitx
